### PR TITLE
[local-path-provisioner] remove wildcard from module RBAC

### PR DESCRIPTION
**Pull request by duckhawk**
## Description

Remove wildcard from module RBAC

## Why do we need it, and what problem does it solve?

We need to reduce the module's permissions to limit available resources.

## Why do we need it in the patch release (if we do)?

This is necessary for some of our customers

## What is the expected result?

More secure platform

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: local-path-provisioner
type: chore
summary: Remove wildcard from module RBAC.
```

### CODE REVIEW

**Review by RomanenkoDenys — APPROVED** [review]

File: modules/031-local-path-provisioner/templates/rbac-for-us.yaml [hunk 1]
```diff
@@ -13,11 +13,11 @@ metadata:
...
 rules:
 - apiGroups: [ "" ]
-  resources: [ "nodes", "persistentvolumeclaims", "configmaps" ]
+  resources: [ "nodes", "persistentvolumeclaims", "configmaps", "pods", "pods/log" ]
   verbs: [ "get", "list", "watch" ]
 - apiGroups: [ "" ]
-  resources: [ "endpoints", "persistentvolumes", "pods" ]
-  verbs: [ "*" ]
+  resources: [ "persistentvolumes" ]
+  verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]
```
Comment (yalosev) [inline modules/031-local-path-provisioner/templates/rbac-for-us.yaml:20]:
```suggestion
  verbs: [ "get", "list", "watch", "create", "patch", "update", "delete", "deletecollection" ]
```

File: modules/031-local-path-provisioner/templates/rbac-for-us.yaml [hunk 2]
```diff
@@ -38,3 +38,29 @@ subjects:
...
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Chart.Name }}
+  namespace: d8-{{ .Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "local-path-provisioner")) | nindent 2 }}
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "watch", "create", "patch", "update", "delete"]
```
Comment (yalosev) [inline modules/031-local-path-provisioner/templates/rbac-for-us.yaml:51]:
```suggestion
    verbs: ["get", "list", "watch", "create", "patch", "update", "delete", "deletecollection"]
```

**Review by name212 — APPROVED** [review]

**Comment by z9r5** [comment]
@duckhawk Matrix tests failed. Please fix.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [8968](https://github.com/deckhouse/deckhouse/pull/8968) to the branch [release-1.62](https://github.com/deckhouse/deckhouse/tree/release-1.62) successful!

**Pull request merged**